crates/oxc_linter/src/rules/eslint/no_empty_static_block.rs

```diff
@@ -0,0 +1,82 @@
+use oxc_ast::AstKind;
+use oxc_diagnostics::{
+    miette::{self, Diagnostic},
+    thiserror::Error,
+};
+use oxc_macros::declare_oxc_lint;
+use oxc_span::Span;
+use regex::Regex;
+
+use crate::{context::LintContext, rule::Rule, AstNode};
+
+#[derive(Debug, Error, Diagnostic)]
+#[error("eslint(no-empty-static-block): Disallow empty static blocks")]
+#[diagnostic(severity(warning), help("Unexpected empty static block."))]
+struct NoEmptyStaticBlockDiagnostic(#[label] pub Span);
+
+#[derive(Debug, Default, Clone)]
+pub struct NoEmptyStaticBlock;
+
+declare_oxc_lint!(
+    /// ### What it does
+    /// Disallows the usages of empty static blocks
+    ///
+    /// ### Why is this bad?
+    /// Empty block statements, while not technically errors, usually occur due to refactoring that wasn’t completed.
+    /// They can cause confusion when reading code.
+    ///
+    /// ### Example
+    /// ```javascript
+    ///
+    /// class Foo {
+    ///     static {
+    ///     }
+    /// }
+    ///
+    /// ```
+    NoEmptyStaticBlock,
+    suspicious
+);
+
+impl Rule for NoEmptyStaticBlock {
+    fn run<'a>(&self, node: &AstNode<'a>, ctx: &LintContext<'a>) {
+        if let AstKind::StaticBlock(static_block) = node.kind() {
+            if static_block.body.is_empty() {
+                let re = Regex::new(r"static(\n|\s|\t)*(//|/[*])").unwrap(); // comments are before the block declaration
+                if ctx.semantic().trivias().has_comments_between(static_block.span)
+                    && !re.is_match(ctx.source_text())
+                {
+                    return;
+                }
+                ctx.diagnostic(NoEmptyStaticBlockDiagnostic(static_block.span));
+            }
+        }
+    }
+}
+
+#[test]
+fn test() {
+    use crate::tester::Tester;
+
+    let pass = vec![
+        "class Foo { static { bar(); } }",
+        "class Foo { static { /* comments */ } }",
+        "class Foo { static {
+			// comment
+			} }",
+        "class Foo { static { bar(); } static { bar(); } }",
+    ];
+
+    let fail = vec![
+        "class Foo { static {} }",
+        "class Foo { static { } }",
+        "class Foo { static {
+
+			 } }",
+        "class Foo { static { bar(); } static {} }",
+        "class Foo { static // comment
+			 {} }",
```

Comment:
do we need the regex because of this case?

Reply:
Yes, I am not sure how we can get the comments between the static and set of braces, otherwise

Reply:
I think since prettier repositions the comment on [format](https://prettier.io/playground/#N4Igxg9gdgLgprEAucAbAhgZ0wAgGIQQ7A6YzowCWYOA9LTpALZMIwA6UO7HvxAvjn4gANCAgAHKtEzJQ6AE4KIAdwAKihLJTpUK9AE9ZYgEYL0YANZwYAZXSsAMpShxkAM12Y4p81Zu2EhYuAObIMAoArj4g3kyU4VExmKGocACKkRDwHl4xAFaYAB62qRlZOUieqN5iAI4VcGrKEtogWAC0rnAAJr2iIBHolKihAMIQLOjI7aioAylQIWkAgjARlCaR8GpwCs6uuTUxABYwTKgA6ieU8JhBYHC2WreUAG63BjNg2ANv0QBJKB9WC2MAKShSFbA2wwAxpI61EASZTeS7mCQzFFwbwKN5uMQuXEwZroEJMaZVPJiIIKXEzEzoExweY0iGwS6UHowE7IAAcAAYxAo4A1KCLSeTKdUkeQTJzubykAAmMSRbwAFSZ2hlMTgTGZPT6PUc6CWkTJcAICgp61CM3Q2wgIH4-CAA). 

we can probably skip this case and remove the regex? What do you think

Reply:
If that is the case, I will remove the regex